libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -875,6 +875,61 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 		m_code << templ.render();
 		break;
 	}
+	case FunctionType::Kind::ABIDecode:
+	{
+		Whiskers templ(R"(
+			let <retVars> := <abiDecode>(<offset>, add(<offset>, add(<offset>, <length>)))
+		)");
+
+		TypePointer firstArgType = arguments.front()->annotation().type;
```

Comment:
```suggestion
		Type const* arrayType = arguments.front()->annotation().type;
```
Or `dataType`? or `encodedType`?